sdk/tables/azure-tables/review/tables.api.review.md

```diff
@@ -0,0 +1,695 @@
+}
+
+// @public
+export type ListEntitiesResponse<T extends object> = Omit<TableQueryEntitiesResponse, "value"> & {
+  readonly value: ListEntitiesIterator<T>;
+};
+
+// @public
+export type ListTablesOptions = Omit<TableQueryOptionalParams, "queryOptions">;
+
+// @public
+export type ListTablesResponse = TableQueryHeaders &
+  TableQueryResponse & {
+    _response: coreHttp.HttpResponse & {
+      bodyAsText: string;
+      parsedBody: TableQueryResponse;
+      parsedHeaders: TableQueryHeaders;
+    };
+  };
+
+// @public
+export interface Logging {
+  delete: boolean;
+  read: boolean;
+  retentionPolicy: RetentionPolicy;
+  version: string;
+  write: boolean;
+}
+
+// @public
+export type MergeEntityResponse = TableMergeEntityHeaders & {
+  _response: coreHttp.HttpResponse & {
+    parsedHeaders: TableMergeEntityHeaders;
+  };
+};
+
+// @public
+export interface Metrics {
+  enabled: boolean;
+  includeAPIs?: boolean;
+  retentionPolicy?: RetentionPolicy;
+  version?: string;
+}
+
+// @public
+export function odata(strings: TemplateStringsArray, ...values: unknown[]): string;
+
+// @public
+export type OdataMetadataFormat =
+  | "application/json;odata=nometadata"
+  | "application/json;odata=minimalmetadata"
+  | "application/json;odata=fullmetadata";
+
+// @public
+export interface QueryOptions extends Omit<GeneratedQueryOptions, "select"> {
+  select?: string[];
+}
+
+// @public
+export type ResponseFormat = "return-no-content" | "return-content";
+
+// @public
+export interface RetentionPolicy {
+  days?: number;
+  enabled: boolean;
+}
+
+// @public
+export interface ServiceGetPropertiesHeaders {
+  clientRequestId?: string;
+  requestId?: string;
+  version?: string;
+}
+
+// @public
+export interface ServiceGetStatisticsHeaders {
+  clientRequestId?: string;
+  date?: Date;
+  requestId?: string;
+  version?: string;
+}
+
+// @public
+export interface ServiceProperties {
+  cors?: CorsRule[];
+  hourMetrics?: Metrics;
+  logging?: Logging;
+  minuteMetrics?: Metrics;
+}
+
+// @public
+export interface ServiceSetPropertiesHeaders {
+  clientRequestId?: string;
+  requestId?: string;
+  version?: string;
+}
+
+// @public
+export type SetAccessPolicyOptions = Omit<TableSetAccessPolicyOptionalParams, "tableAcl">;
+
+// @public
+export type SetAccessPolicyResponse = TableSetAccessPolicyHeaders & {
+  _response: coreHttp.HttpResponse & {
+    parsedHeaders: TableSetAccessPolicyHeaders;
+  };
+};
+
+// @public
+export interface SetPropertiesOptions extends coreHttp.OperationOptions {
+  requestId?: string;
+  timeout?: number;
+}
+
+// @public
+export type SetPropertiesResponse = ServiceSetPropertiesHeaders & {
+  _response: coreHttp.HttpResponse & {
+    parsedHeaders: ServiceSetPropertiesHeaders;
+  };
+};
+
+// @public
+export interface SignedIdentifier {
+  accessPolicy: AccessPolicy;
+  id: string;
+}
+
+// @public
+export class TableClient {
+  constructor(url: string, tableName: string, options?: TableServiceClientOptions);
```

Comment:
WIP: we're working on a constructor override that takes a SharedKeyCredential.

```typescript
constructor(url: string, tableName: string, credentials: SharedKeyCredential)
```

Client creation
```typescript
const tableClient= new TableClient("accountUrl", "TestTable", new SharedKeyCredential("accountName", "accountKey"));
```